### ISSUE TITLE
Collection hierarchy diagrams in collections guides are out of date

### ISSUE THREAD

**Issue by SethTisue** [scala/docs.scala-lang#1360]
They were update at some point in the 2.13 cycle, but they aren't exactly right for 2.13.0, for example we noticed that `TreeSeqMap` and `CollisionProofHashMap` were missing.

**Comment by Philippus** [comment]
I think @mslinn is working on this?
https://github.com/mslinn/scala-collections-charts/tree/2.13

**Comment by mslinn** [comment]
The reworked collection charts are ready for review at https://github.com/mslinn/scala-collections-charts/tree/2.13

**Comment by NthPortal** [comment]
should they include `IterableOnce`?

**Comment by mslinn** [comment]
I don't think that end users should care about an implementation detail

**Comment by NthPortal** [comment]
`IterableOnce` is a trait (implemented by `Option`, no less, though I'm not sure I'd put that in the diagram)

**Comment by mslinn** [comment]
The goal of this project is merely to provide end users an intuitive understanding of the collection classes. Conceptually `IterableOnce`  is a [distinction without a difference](https://en.wikipedia.org/wiki/Distinction_without_a_difference) to everyone except to library authors of collection classes. I will add a description of the goals and non-goals of this project to the README.

**Comment by dwijnand** [comment]
Isn't whether a collection can be iterated only once or multiple times an important distinction?

**Comment by mslinn** [comment]
These charts are not intended to replace the need for written documentation. If it would be possible to include that level of detail in the charts, and keep them easy to read, that would be fine. If this is the only (deliberate) omission, then lets include it. Before making that decision, lets consider what else might be included with that level of detail.

Alternatively, it might be possible to use UML annotation instead to show 0..1 iterations, but again that would required lots of fine print and I would argue against an accurate but confusing and mostly illegible set of charts.
